source/routing/next_stop_time.cpp

```diff
@@ -338,6 +341,126 @@ NextStopTime::tardiest_stop_time(const StopEvent stop_event,
     return first_discrete_st_pair;
 }
 
+
+template<typename F>
+static void vj_loop(const nt::DiscreteVehicleJourney*, F f) {
+    // Discrete Vehicle Journey doesn't need to loop
+    f(0);
+}
+
+template<typename F>
+static void vj_loop(const nt::FrequencyVehicleJourney* vj, F f) {
+    for (auto freq_shift = vj->start_time; freq_shift <= vj->end_time; freq_shift+= vj->headway_secs) {
+        f(freq_shift);
```

Comment:
so now we expand the frequency vj ?

We need to bench this on datasets with ODT to check the differences.